packages/features/auth/lib/next-auth-options.ts

```diff
@@ -944,6 +944,12 @@ export const getOptions = ({
                 identityProviderId: account.providerAccountId,
               },
             });
+
+            if (existingUserWithEmail.twoFactorEnabled) {
```

Comment:
when `existingUserWithEmail` is true, we either route user to `auth/error/...` page or get user logged in except this particular conditional.